python/ray/_raylet.pyx

```diff
@@ -558,6 +562,7 @@ cdef CRayStatus task_execution_handler(
 
     with gil:
         try:
+            client_was_enabled = _disable_client_hook()
```

Comment:
When is this needed? Won't the client always be disabled on the server?

Reply:
Same as above -- especially in local mode.

In fact, local mode is exactly the reason to consider this approach. Local mode expects much more about running in process.